Tests/stm32_test.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+from time import sleep
+import pytest
+import serial
+import sys
+import os
+
+#Setup serial comm
+ser = serial.Serial(
+        port='/dev/ttyAMA0',
+        baudrate=115200,
+        parity=serial.PARITY_NONE,
+        stopbits=serial.STOPBITS_ONE,
+        bytesize=serial.EIGHTBITS,
+        timeout=0.1) 
+
+
+COMMAND_MALFORMAT = b'Command does not match format.\n'
+COMMAND_OK = b'OK.\n'
+COMMAND_NOT_RECOGNISED = b'Command not recognised.\n'
+
+#Send command to MCU and return last line read via UART
+#Function also prints each line received
+def parse_command(command):
+    print("Testing: ",command.decode('ascii'),end='')
+    ser.write(command)
+    answer = bytearray()
+    answer_byte = ser.read()
+    while(isinstance(answer_byte, bytes) and answer_byte != b''):
+        answer.extend(answer_byte)
+        answer_byte = ser.read()
+    answer_line = bytearray()
+    char_counter = 0
+    for char in answer:
+        char_counter = char_counter + 1
+        answer_line.append(char)
+        if chr(char) == '\n':
+            print(answer_line.decode('ascii'),end='')
+            # Clear line unless it is the last line
+            if(char_counter < len(answer)):
+                answer_line.clear()
+    return answer_line
+
+# String commands to be tested
+@pytest.mark.parametrize("string_command,string_result", [
```

Comment:
How hard is it to have these command <-> response pairs in a separate file and pulled in when code executes?